File: inverted_pendulum_controls/scripts/gazebo_tutorial_listener.py
#!/usr/bin/env python

import rospy
from controls.msg import inverted_pendulum_states


def chatterCallback(msg):
    torque = msg.wheel_torques
    angle = msg.pendulum_angle
    angle_velocity = msg.pendulum_angle_velocity
    speed = msg.chasis_speed
    rospy.loginfo("%f,%f,%f,%f\n", torque, angle, angle_velocity, speed)


def listener():
    rospy.init_node('listener', anonymous=True)
    rospy.Subscriber("pendulum_states",
                     inverted_pendulum_states, chatterCallback)
    rospy.spin()


if __name__ == "__main__":
    listener()

File: inverted_pendulum_controls/src/gazebo_tutorial_listener.cpp
#include "ros/ros.h"
#include "controls/inverted_pendulum_states.h"


void chatterCallback(const controls::inverted_pendulum_states& msg)
{
  float torque = msg.wheel_torques;
  float angle = msg.pendulum_angle;
  float angle_velocity = msg.pendulum_angle_velocity;
  float speed = msg.chasis_speed;
  ROS_INFO("%f,%f,%f,%f\n", torque,angle,angle_velocity,speed);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("pendulum_states", 1000, chatterCallback);
  ros::spin();
  return 0;
}
